src/parsers/installrdf.js

```diff
@@ -1,66 +1,77 @@
-import { ADDON_TYPE_MAP, INSTALL_RDF, RDF_DEFAULT_NAMESPACE } from 'const';
-import { TYPE_INVALID, TYPE_MISSING } from 'messages';
+import { ADDON_TYPE_MAP, RDF_DEFAULT_NAMESPACE } from 'const';
+import { RDF_TYPE_INVALID, RDF_NAME_MISSING,
+         RDF_TYPE_MISSING } from 'messages';
 import log from 'logger';
-import RDFScanner from 'scanners/rdf';
 
 
 export default class InstallRdfParser {
 
-  constructor(rdfString, collector, {namespace=RDF_DEFAULT_NAMESPACE}={}) {
-    this.rdfString = rdfString;
+  constructor(xmlDoc, collector, {namespace=RDF_DEFAULT_NAMESPACE}={}) {
+    this.xmlDoc = xmlDoc;
     // Provides ability to directly add messages to
     // the collector.
     this.collector = collector;
     this.namespace = namespace;
   }
 
-  parseDoc() {
-    var rdfScanner = new RDFScanner(this.rdfString, INSTALL_RDF);
-    return rdfScanner.getContents();
-  }
-
   getMetaData() {
-    return this.parseDoc()
-      .then((xmlDoc) => {
-        return Promise.resolve({
-          guid: this._getGUID(xmlDoc),
-          type: this._getAddonType(xmlDoc),
-        });
-      });
+    return Promise.resolve({
+      guid: this._getGUID(),
+      name: this._getName(),
+      type: this._getAddonType(),
+    });
   }
 
-  _getAddonType(xmlDoc) {
-    var addonType = null;
-    var typeNodes = xmlDoc.getElementsByTagName('em:type');
+  _getNode(tag) {
+    var typeNodes = this.xmlDoc.getElementsByTagName(tag);
     if (typeNodes.length > 1) {
-      throw new Error('Multiple <em:type> elements found');
+      throw new Error(`Multiple <${tag}> elements found`);
```

Comment:
Are there no scenarios where multiple tags of the same type are allowed? I think there are instances where the same tag appears multiple times and it's okay.